packages/i18n-calypso/src/use-translate.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+export default function( i18n ) {
+	const translate = i18n.translate.bind( i18n );
+	const getLocaleSlug = i18n.getLocaleSlug.bind( i18n );
+
+	return function useTranslate() {
+		const [ localeSlug, setLocaleSlug ] = React.useState( getLocaleSlug );
```

Comment:
See https://reactjs.org/docs/hooks-reference.html#lazy-initial-state for what passing a function to useState means.

Reply:
Yes, that's exactly what we need here, isn't it? The first render will initialize the state value, and all further changes come from the `change` listener.

Reply:
yup! I just found it surprising, as I didn't know that useState could take an initializer function. Sorry I wasn't more clear.

Reply:
my first comment should have read: "A note to anyone else reviewing this PR, see ..."

Reply:
Here's a great blog post that describes many possible gotchas: https://overreacted.io/making-setinterval-declarative-with-react-hooks/

One source of subtle bugs is capturing a stale value inside a closure:
```js
const [ count, setCount ] = useState( 0 );
useEffect( () => {
  setTimeout( () => setCount( count + 1 ), 1000 );
}, [] );
```
Because of the `[]` argument, `useEffect` will call the effect function only once, on mount, and the `count` value will remain bound to `0`, even if the state changed in the meantime.

It's interesting that the `setCount` function doesn't have this problem and can be saved without problems. It's assumed to be the same for every call. Ref objects returned from `useRef()` can also be saved. I don't know what exactly the rules and guarantees are -- which values are safe and which are not?